core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/ModeStateMachineServices.java

```diff
@@ -170,6 +175,61 @@ public EObject showHideStatesInStateAndModeDiag(DSemanticDecorator view, List<St
     return view;
   }
 
+  public List<StateTransition> getVirtualTransitions(Region region, DDiagram diagram) {
+    // Get all transitions that should be displayed in that diagram
+    ECrossReferenceAdapter crossRef = Session.of(region).get().getSemanticCrossReferencer();
+    TreeIterator<EObject> contentIte = region.eAllContents();
+    List<StateTransition> transitions = new ArrayList<>();
+    while (contentIte.hasNext()) {
+      EObject next = contentIte.next();
+
+      if (next instanceof StateTransition) {
+        StateTransition transition = (StateTransition) next;
+
+        AbstractState source = transition.getSource();
+
+        AbstractDNode sourceRep = getMappingMSMStateMode(crossRef, source, diagram);
+        if (sourceRep != null && !sourceRep.isVisible()) {
```

Comment:
here, i think the filter doesn't work when the user use an "unsynchronized" diagram.
In an unsyncrhoznied diagram, all the sub modes are not displayed automatically but can be inserted/removed by the user.

(in capella tests, and more generally in capella code, the "show/hide keywords" reflects this behavior (more an insert/remove) and not the show/hide sirius feature which is using the setVisible (i.e. the MSMdiagram.hideState will "remove" the mode from the diagram, not "sirius hide" it.)

Reply:
Create some states with an internal transition
![image](https://user-images.githubusercontent.com/34767327/149529014-805661b7-2012-4986-b7e4-dff04afd46bb.png)

Mark diagram as unsynchronized
![image](https://user-images.githubusercontent.com/34767327/149529063-39f5c043-6384-4487-8354-236d0acd8dd4.png)

Delete inner modes from diagram, 
- Using the tool "Show/Hide StateMode" and moving left both sub modes in the wizard.
![image](https://user-images.githubusercontent.com/34767327/149529969-7b9d095b-6fab-44c6-b8b3-39dae9258aa3.png)
- or you can use the "one-red-cross" button in the toolbar (it has the same effect)
![image](https://user-images.githubusercontent.com/34767327/149529756-af9a9083-83fe-4b1f-910b-e2b3d873ee35.png)

Refresh, the computed link shall appear, even if the sub DNodes and DEdges no longer exists in the diagram (sourceRep /target are null)

Reply:
I have tried the same use case with LAB and component exchanges. During the refresh, the computed exchanges don't appear with an unsynchronised diagram, only with a synchronised diagram. Could you confirm this behavior ?
When I look at the DDiagramSychronizer, DEdgeCandidate are only created if the diagram is Synchronized : org.eclipse.sirius.diagram.business.internal.sync.DDiagramSynchronizer.refreshEdgeMapping(DiagramMappingsManager, Map<DiagramElementMapping, Collection<EdgeTarget>>, EdgeMapping, Map<EdgeMapping, Collection<MappingBasedDecoration>>, Map<String, Collection<SemanticBasedDecoration>>, IProgressMonitor)
![image](https://user-images.githubusercontent.com/3184827/149533109-87ef3208-fa66-438e-b472-cf474dd77375.png)
Is there something I'm missing ?